pkg/cmd/cli/cmd/edit.go

```diff
@@ -59,14 +78,24 @@ saved copy to include the latest resource version.`
 
 // NewCmdEdit implements the OpenShift cli edit command
 func NewCmdEdit(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
-	filenames := []string{}
+	var options EditOptions
+	options.fullName = fullName
 	cmd := &cobra.Command{
 		Use:     "edit (RESOURCE/NAME | -f FILENAME)",
 		Short:   "Edit a resource on the server",
 		Long:    editLong,
 		Example: fmt.Sprintf(editExample, fullName),
 		Run: func(cmd *cobra.Command, args []string) {
-			err := RunEdit(fullName, f, out, cmd, args, filenames)
+			options.cmd = cmd
+			options.args = args
+
+			if err := options.Complete(); err != nil {
```

Comment:
Complete should accept all those arguments you have at RunEdit. Well, all those that are useful to it.